my-app/src/Components/Navigation.tsx

```diff
@@ -86,13 +71,27 @@ export const Navigation = () => {
 
    const onAuthClick = () => {
       setPage('auth')
+   }
+
+   const onSearchClick = (inputValue: string) => {
+
+      if (page !== 'auth' && page !== 'reg') {
+         setPage('searchPosts')
+         setSearchInputValue(inputValue)
+         console.log(inputValue)
+      }
 
    }
 
+   document.body.style.backgroundColor = theme === 'light' ? '#F3F3F3' : '#141414';
+
    return <>
+      <Header username={(page !== 'auth' && page !== 'reg') ? user.username : null} clickSearch={onSearchClick} />
       {page === 'auth' && <Auth onAuth={onAuth} onRegClick={onRegClick} />}
       {page === 'reg' && <Registration onReg={onReg} onAuthClick={onAuthClick} />}
-      {page === 'postsList' && <PostsList username={user.username} onPostClick={onPostClick} />}
-      {selectedPost && <PostPage username={user.username} post={selectedPost} />}
+      {page === 'postsList' && <PostsList onPostClick={onPostClick} />}
+      {page === 'searchPosts' && <SearchPostsList searchInputValue={searchInputValue} onPostClick={onPostClick}></SearchPostsList>}
```

Comment:
there is no need in `SearchPostsList`
you have the same component PostsList, that can be rendered with different data